scipy/optimize/_elementwise.py

```diff
@@ -799,3 +827,96 @@ def bracket_minimum(f, xm0, *, xl0=None, xr0=None, xmin=None, xmax=None,
     del res.fm
     del res.fr
     return res
+
+
+def _secant(func, a, b, *, args=(), xatol=None, xrtol=None,
+            fatol=None, frtol=0, maxiter=None, callback=None):
+    res = _chandrupatla_iv(func, args, xatol, xrtol,
+                           fatol, frtol, maxiter, callback)
+    func, args, xatol, xrtol, fatol, frtol, maxiter, callback = res
+
+    # Initialization
+    temp = eim._initialize(func, (a, b), args)
+    func, xs, fs, args, shape, dtype, xp = temp
+    x1, x2 = xs
+    f1, f2 = fs
+    status = xp.full_like(x1, xp.asarray(eim._EINPROGRESS),
+                          dtype=xp.int32)  # in progress
+    nit, nfev = 0, 2  # two function evaluations performed above
+    finfo = xp.finfo(dtype)
+    xatol = 4*finfo.smallest_normal if xatol is None else xatol
+    xrtol = 4*finfo.eps if xrtol is None else xrtol
+    fatol = finfo.smallest_normal if fatol is None else fatol
+    frtol = frtol * xp.minimum(xp.abs(f1), xp.abs(f2))
+    maxiter = (math.log2(finfo.max) - math.log2(finfo.smallest_normal)
+               if maxiter is None else maxiter)
+    work = _RichResult(x1=x1, f1=f1, x2=x2, f2=f2, x3=None, f3=None, t=0.5,
+                       xatol=xatol, xrtol=xrtol, fatol=fatol, frtol=frtol,
+                       nit=nit, nfev=nfev, status=status)
+    res_work_pairs = [('status', 'status'), ('x', 'xmin'), ('fun', 'fmin'),
+                      ('nit', 'nit'), ('nfev', 'nfev'), ('xl', 'x1'),
+                      ('fl', 'f1'), ('xr', 'x2'), ('fr', 'f2')]
+
+    def pre_func_eval(work):
```

Comment:
@theDDA Please take a look at how this works and think about how Muller's method would fit in.
- `pre_func_eval` figures out the new abscissa evaluation point based on the state
- `post_func_eval` updates the state based on the function evaluation
- `check_termination` determines which, if any, stopping conditions are valid
- `post_termination_check` is anything that might need to be done after the termination check but before the next iteration's `pre_func_eval`
- `customize_result` customizes the result object provided to the user as the input to the callback function or the return of `find_root`. Otherwise, the user just gets what is specified in `res_work_pairs` (attributes that should be in the result given the value that is in `work`)